extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/BasicSecuritySSLSocketFactory.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.security.basic;
+
+import org.apache.druid.java.util.common.logger.Logger;
+
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+
+public class BasicSecuritySSLSocketFactory extends SSLSocketFactory
+{
+  private static final Logger LOG = new Logger(BasicSecuritySSLSocketFactory.class);
+  private SSLSocketFactory sf;
+
+  public BasicSecuritySSLSocketFactory()
+  {
+    try {
+      SSLContext ctx = SSLContext.getInstance("TLS");
```

Comment:
The SSLContext should be exposed for configuration so users can set the keystore, truststore, protocol, etc.

`TLSUtils` has some examples for doing that, used by `SSLContextProvider` for the internal HTTP client

Reply:
thanks for this info, I'll look into it and make updates today

Reply:
I've addressed this, SSLContext is exposed for configuration and uses TLSUtils to build the SSLContext